src/pages/images-constructor/tests/images-constructor.spec.js

```diff
@@ -36,28 +36,36 @@ const mockUseDispatch = jest.spyOn(redux, 'useDispatch');
 mockUseDispatch.mockImplementation(() => mockDispatch);
 
 beforeEach(async () => {
-  mockUseDispatch.mockImplementation(() => mockDispatch);
-  render(
-    <MockedProvider mocks={mockAllConstructors} addTypename={false}>
-      <ImagesConstructor />
-    </MockedProvider>
-  );
-  await new Promise((resolve) => setTimeout(resolve, 2000));
+  await act(async () => {
```

Comment:
This is a bad practice to use act like that. This only works because your operation happens in the next tick of the event loop. And act() here basically just hide warning about "React state update shold be wrapped..."

As a solution you can remove useless act and await new Promise and add await before waitForElement

Reply:
Done.